src/main/java/com/example/gamelist/GameService.java

```diff
@@ -0,0 +1,26 @@
+package com.example.gamelist;
+
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class GameService {
+
+    private final GameMapper gameMapper;
+
+    public GameService(GameMapper gameMapper) {
+        this.gameMapper = gameMapper;
+    }
+
+    public void updateGame(final Integer id, String name) {
+        Game game = gameMapper.findById(id).orElseThrow(() -> new GameNotFoundException("game information not found"));
+        List<Game> otherGames = this.gameMapper.findByName(name);
+
+        if (otherGames.stream().anyMatch(game -> game.getName().equals(name))) {
+            game.setName(UpdateRequest.getName());
+        } else throw new DuplicateKeyException("Already registered data");
```

Comment:
これも間違いではないですし、いいのですが、
バグの原因になりかねないので { } つけておきましょうか。
```suggestion
        } else {
            throw new DuplicateKeyException("Already registered data");
        }
```

Reply:
34b6bb543d4df47c0cc2b413f1b9d050a2df5085
修正行いました。

Reply:
OKです。